usaspending_api/api_contracts/contracts/v2/disaster/agency/loans.md

```diff
@@ -0,0 +1,127 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# Agencies Spending Disaster/Emergency Funding via Loans [/api/v2/disaster/agency/loans/]
+
+This endpoint provides insights on the Agencies awarding loans from disaster/emergency funding per the requested filters.
+
+## POST
+
+Returns loan spending details of Agencies receiving supplemental funding budgetary resources
+
++ Request (application/json)
+    + Schema
+
+            {
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object"
+            }
+
+    + Attributes
+        + `filter` (required, Filter, fixed-type)
+        + `pagination` (optional, Pagination, fixed-type)
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + `results` (required, array[Result], fixed-type)
+        + `pagination_metadata` (required, PageMetadata, fixed-type)
+
+
+    + Body
+
+            {
+                "results": [
+                    {
+                        "id": 43,
+                        "code": "090",
+                        "description": "Description text of 090, for humans",
+                        "children": [],
+                        "count": 54,
+                        "face_value_of_loan": 89.01
+                    },
+                    {
+                        "id": 41,
+                        "code": "012",
+                        "description": "Description text of 012, for humans",
+                        "children": [],
+                        "count": 2,
+                        "face_value_of_loan": 50
+                    }
+                ],
+                "pagination_metadata": {
+                    "page": 1,
+                    "next": 2,
+                    "previous": null,
+                    "hasNext": true,
+                    "hasPrevious": false,
+                    "total": 23,
+                    "limit": 2
+                }
+            }
+
+# Data Structures
+
+## Filter (object)
```

Comment:
In general, do we ever try to name these objects uniquely? e.g. a `Filter` object for this endpoint is a different structure for similarly named `Filter` objects in other endpoints. Should we distinguish difference _kinds_ of `Filter` objects by their name, e.g. `DisasterAgencyLoanFilter`? Or is it implied to be "namespaced" in the context of this one endpoint, e.g. `disaster.agency.loans.Filter`. Just curious.

Reply:
Since importing other files isn't native to markdown, each file is its own namespace. There is a small effort to use slightly less generic names for identical objects across various files which need to be the same because of shared API code. I'm unsure how these endpoints will share code, but it might make sense to use a less generic name for these objects since they will need to remain consistent across the APIs regardless of the implementation.